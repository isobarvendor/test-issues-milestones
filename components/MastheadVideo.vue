<template>
  <div id="masthead" class="container" >

 <video-background
    :src="videoDesk"
    :poster="imgDesk"
    :sources="[
        {src: videoDesk, res: 1100, autoplay: true},
        {src: videoTab, res: 900, autoplay: true},
        {src: videoMob, res: 638, autoplay: true}
    ]"
    class="videoBackground"
    style="height: 100vh;"
    overlay="linear-gradient(45deg,#2a4ae430,#fb949e6b)"
>
      <div class="masthead-content">
      <h1>{{data.title}} testing</h1>
      <p>{{data.description}} hello hellohello hellohello hellohello hellohello hellohello hellohello hellohello hellohello hellohello hellohello hello</p>
    <vac :end-time="new Date().getTime() + 60000" @finish="finishCall">
    <span
      slot="process"
      slot-scope="{ timeObj }">{{ `Lefttime: ${timeObj.m}:${timeObj.s}` }}</span>
    <span slot="finish">Done!</span>
  </vac>
    </div>
</video-background>

  </div>
</template>

<script>
export default {
  name: "MastheadVideo",
  props: {
    data: null
  },
  data() {
    return {
      imgDesk: this.data.desktopImage[0].url,
      imgTab: this.data.tabletImage[0].url,
      imgMob: this.data.mobileImage[0].url,
      videoDesk: "/img/background.mp4",
      videoTab:  "/img/background.mp4",
      videoMob:  "/img/background.mp4",
    }
  },
  mounted() {

  },
  methods: {
    finishCall(){
      alert('tes')
    }
  },
  mounted() {

  },
};
</script>

<style>
  @media only screen and (max-width: 600px) {
    .videoBackground{
      max-height: 400px !important;
    }
  }
    @media only screen and (max-width: 1099px) {
    .videoBackground{
      max-height: 500px;
    }
  }
      @media only screen and (min-width: 1100px) {
    .videoBackground{
      max-height: 600px;
    }
  }
</style>
