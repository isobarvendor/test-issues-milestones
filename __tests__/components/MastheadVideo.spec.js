
import { mount } from '@vue/test-utils'

import MastheadVideo from "@/components/MastheadVideo";


const factory = () => {
  return mount(MastheadVideo, {

    propsData: {
      data :{
              mobileImage:[{url:"mobile image"}],
              desktopImage:[{url:"desktop image"}],
              tabletImage:[{url:"tablet image"}],
            }
    }
  });
};

describe("MastheadVideo", () => {





  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
