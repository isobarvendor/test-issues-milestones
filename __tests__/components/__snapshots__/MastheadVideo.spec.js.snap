// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MastheadVideo renders properly 1`] = `
<div id="masthead" class="container">
  <section data-v-7bbc3657="" class="vue-responsive-videobg videoBackground" style="height: 100vh;">
    <div data-v-b5717866="" data-v-7bbc3657="" class="video-buffering"></div>
    <transition-stub data-v-229e82cd="" data-v-7bbc3657="" name="fade">
      <div data-v-229e82cd="" class="video-wrapper" style="display: none;"><video data-v-229e82cd="" autoplay="autoplay" playsinline="" loop="loop" preload="auto" style="object-fit: cover;">
          <source data-v-229e82cd="" src="/img/background.mp4" type="video/mp4">
        </video></div>
    </transition-stub>
    <div data-v-ea6e5e6a="" data-v-7bbc3657="" class="video-overlay"></div>
    <div data-v-7bbc3657="" class="videobg-content">
      <div data-v-7bbc3657="" class="masthead-content">
        <h1 data-v-7bbc3657=""> testing</h1>
        <p data-v-7bbc3657=""> hello hellohello hellohello hellohello hellohello hellohello hellohello hellohello hellohello hellohello hellohello hello</p>
      </div>
    </div>
  </section>
</div>
`;
